sklearn/metrics/_ranking.py

```diff
@@ -375,12 +375,11 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None, max_fpr=None):
         warnings.warn(
             (
                 "Only one class is present in y_true. ROC AUC score "
-                "is not defined in that case. The score is set to "
-                "0.0."
+                "is not defined in that case."
```

Comment:
It would be helpful to store `np.unique(y_true)` in a local variable and then display the value of that class label in the warning message.

Reply:
To my understanding and exploration of the code, `y_true` can already be swapped at this point, so the warning would be misleading. (See https://github.com/scikit-learn/scikit-learn/issues/30079#issuecomment-2423990355).

I would prefer to keep this as it is, but if anybody wants to change it - go ahead. :)

Reply:
I think the idea here is different: we are not going to say that it is a positive or a negative class, we will just show the value that we found, i.e
```
f"Only one class is present in y_true: {y_true[0]!r}. "
f"ROC AUC score is not defined in that case."
```
Or something like that.

Reply:
Please let us stop this.

As I wrote, classes are renumerated. I know next to nothing about sklearn's code, but if caller gives `y_true=[5, 2, 5]` to `roc_auc_score`, the `y_true` that is passed to this internal function will be `[1, 0, 1]`. If caller gives `[1, 1, 1]`, this function gets `[0, 0, 0]`. Any  `y_true` composed of equal values is renumerated to `y_true`'s composed of 0s.

If I change the function to

```python
def _binary_roc_auc_score(y_true, y_score, sample_weight=None, max_fpr=None):
    if len(np.unique(y_true)) != 2:
        print("all values equal", y_true[0])
    return
```

I get the following

```
>>> import sklearn.metrics
sklearn.metrics.roc_auc_score([0, 0, 0], [0.3, 0.4, 0.5])
all values equal 0
>>> sklearn.metrics.roc_auc_score([1, 1, 1], [0.3, 0.4, 0.5])
all values equal 0
>>> sklearn.metrics.roc_auc_score([5, 5, 5], [0.3, 0.4, 0.5])
all values equal 0
```


To my possibly incomplete understanding of the code, `len(np.unique(y_true)) != 2` may be equivalent to `not np.any(t_true)`.

This function cannot give a more descriptive warning, because it doesn't have sufficient information for it. A better warning could be given by the function that calls it, but it would require a big refactoring because a (curried) `_binary_roc_auc_score` is given as an argument to `_average_binary_score`.

Changing a single `0.0` into `np.nan` and adding two lines of tests has taken me several hours spread across five days, mostly because of these warnings that could, imho, stay as they were. Not a great experience that I would want to repeat. :)